puppet/zulip/manifests/profile/smokescreen.pp

```diff
@@ -34,13 +34,22 @@
     creates     => $bin,
     require     => [File['/srv/golang'], Zulip::Sha256_tarball_to['smokescreen']],
   }
-
+  file { $bin:
+    ensure  => file,
+    require => Exec['compile smokescreen'],
+  }
   file { '/usr/local/bin/smokescreen':
     ensure  => 'link',
     target  => $bin,
-    require => Exec['compile smokescreen'],
+    require => File[$bin],
     notify  => Service[supervisor],
   }
+  tidy { '/usr/local/bin/smokescreen-*':
```

Comment:
How exactly does `tidy` work?  I couldn't quickly confirm in the puppet documentation that it doesn't delete resources declared elsewhere, but I assume that's the intent.

Reply:
It doesn't delete resources that are declared elsewhere, which is why `file { $bin: ... }` was added right above.

Reply:
I can't find a citation for it in the docs, but that's the observed behaviour, and:
https://github.com/puppetlabs/puppet/blob/c28e71c38d26c17b580d9eb672733e67a79a17c7/lib/puppet/type/tidy.rb#L311-L315

Reply:
OK.  It seems pretty clearly to be the intent of the `tidy` feature, but it's crazy they don't document it.

Reply:
(Maybe worth opening a quick issue upstream?)

Reply:
Filed https://tickets.puppetlabs.com/browse/PUP-11367